x509/src/der_helper.rs

```diff
@@ -0,0 +1,102 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    der_helper.rs
+
+Abstract:
+
+    Helpers for encoding DER unsigned integers
+
+--*/
+
+/// DER Integer Tag
+const DER_INTEGER_TAG: u8 = 0x02;
+
+#[inline(never)]
+fn subslice_without_zero(val: &[u8]) -> &[u8] {
+    // Count the leading zeros
+    let clz = val
+        .iter()
+        .enumerate()
+        .find(|(_, byte)| **byte != 0)
+        .map_or(val.len(), |(idx, _)| idx);
+
+    // If everything is 0, then we need len 1, and 0 as value
+    val.get(clz..).unwrap_or(&[0])
```

Comment:
Should this be `unwrap_or(&val[0]))`

Also of `val` is empty we may want to fail.

Reply:
This is not new behavior. I suppose we don't have a use case for '0' encoding?

Reply:
test_rt_wdt_timeout fails if we don't encode 0 correctly.